source/wagi/WasiExperimentalHttp/HttpRequestHandler.cs

```diff
@@ -0,0 +1,459 @@
+﻿namespace Wasi.Experimental.Http
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Globalization;
+  using System.IO;
+  using System.Linq;
+  using System.Net.Http;
+  using System.Net.Http.Headers;
+  using System.Text;
+  using System.Threading;
+  using Microsoft.Extensions.Logging;
+  using Wasi.Experimental.Http.Exceptions;
+  using Wasmtime;
+
+  /// <summary>
+  /// HttpRequestHandler provides support for wasi_experimental_http.
+  /// </summary>
+  internal class HttpRequestHandler : IDisposable
+  {
+    private const string ModuleName = "wasi_experimental_http";
+    private const string MemoryName = "memory";
+    private const int MaxResponses = 10;

```

Comment:
The default behavior in the Rust implementation is that if the user does not set a maximum here, it is only limited by the underlying platform (i.e. when the host runs out of outbound connections, for example).
Do we want to enforce a number here?

Reply:
OK, I will make it configurable via config settings and default it to a higher value.